src/openzeppelin/token/erc1155/library.cairo

```diff
@@ -0,0 +1,601 @@
+
+        _do_safe_batch_transfer_acceptance_check(
+            operator, from_, to, ids_len, ids, amounts_len, amounts, data_len, data
+        );
+        return ();
+    }
+
+    func _mint{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+        to: felt, id: Uint256, amount: Uint256, data_len: felt, data: felt*
+    ) {
+        // Cannot mint to zero address
+        with_attr error_message("ERC1155: mint to the zero address") {
+            assert_not_zero(to);
+        }
+        // Check uints validity
+        _check_id(id);
+        with_attr error_message("ERC1155: amount is not a valid Uint256") {
+            uint256_check(amount);
+        }
+
+        // add to minter, check for overflow
+        _add_to_receiver(id, amount, to);
+
+        // Emit events and check
+        let (operator) = get_caller_address();
+        TransferSingle.emit(operator=operator, from_=0, to=to, id=id, value=amount);
+        _do_safe_transfer_acceptance_check(
+            operator=operator, from_=0, to=to, id=id, amount=amount, data_len=data_len, data=data
+        );
+        return ();
+    }
+
+    func _mint_batch{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+        to: felt,
+        ids_len: felt,
+        ids: Uint256*,
+        amounts_len: felt,
+        amounts: Uint256*,
+        data_len: felt,
+        data: felt*,
+    ) {
+        alloc_locals;
+        // Cannot mint to zero address
+        with_attr error_message("ERC1155: mint to the zero address") {
+            assert_not_zero(to);
+        }
+        // Check args are equal length arrays
+        with_attr error_message("ERC1155: ids and amounts length mismatch") {
+            assert ids_len = amounts_len;
+        }
+
+        // Recursive call
+        _mint_batch_iter(to, ids_len, ids, amounts);
+
+        // Emit events and check
+        let (operator) = get_caller_address();
+        TransferBatch.emit(
+            operator=operator,
+            from_=0,
+            to=to,
+            ids_len=ids_len,
+            ids=ids,
+            values_len=amounts_len,
+            values=amounts,
+        );
+        _do_safe_batch_transfer_acceptance_check(
+            operator=operator,
+            from_=0,
+            to=to,
+            ids_len=ids_len,
+            ids=ids,
+            amounts_len=amounts_len,
+            amounts=amounts,
+            data_len=data_len,
+            data=data,
+        );
+        return ();
+    }
+
+    func _burn{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+        from_: felt, id: Uint256, amount: Uint256
+    ) {
+        alloc_locals;
+        with_attr error_message("ERC1155: burn from the zero address") {
+            assert_not_zero(from_);
+        }
+
+        // Check uints validity
+        _check_id(id);
+        with_attr error_message("ERC1155: amount is not a valid Uint256") {
```

Comment:
If we have a `_check_id`, maybe is worth adding a `_check_amount` for consistency (we have 6 places where this assertion happens).